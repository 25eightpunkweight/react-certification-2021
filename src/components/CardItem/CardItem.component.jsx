import React from 'react';
import styled from 'styled-components';

// import { useFortune } from '../../utils/hooks/useFortune';
// import './CardItem.styles.css';
import ImgWrap from '../ImgWrap';
import CardTitle from '../CardTitle';
import ImageHover from '../ImageHover';

const CardItem = styled.div`
  height: 330px;
  background-color: white;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin: 5px;
  flex-basis: calc(20% - 22px);
  position: relative;
`;

function Component(item) {
  return (
    // <div className="card-item" key={item.etag}>
    //   <div className="card-title">
    //     <h4>{item.snippet.title}</h4>
    //   </div>
    //   <div className="img__wrap">
    //     <img className="center" src={item.snippet.thumbnails.medium.url} />
    //     <p className="img__description">{item.snippet.description}</p>
    //   </div>
    // </div>

    <CardItem key={item.etag}>
      <CardTitle text={item.snippet.title} />
      <ImgWrap
        image={item.snippet.thumbnails.medium.url}
        hover_text={item.snippet.description}
      />
    </CardItem>
  );
}

export default Component;
