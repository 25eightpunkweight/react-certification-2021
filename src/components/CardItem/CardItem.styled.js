import styled from 'styled-components';

const CardItemDiv = styled.div`
  height: 330px;
  background-color: ${(props) => (props.theme.darkMode ? `#000` : `white`)};
  border-radius: 5px;
  border: 1px solid #ddd;
  margin: 5px;
  flex-basis: calc(20% - 22px);
  position: relative;
`;

const Styled = {
  CardItemDiv,
};

export default Styled;
