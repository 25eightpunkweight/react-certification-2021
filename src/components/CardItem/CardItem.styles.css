.header {
  text-align: center;
  color: white;
  margin: 0;
  height: 3em;
  background-color: black;
}

div.card-item {
  height: 300px;
  background-color: white;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin: 5px;
  flex-basis: calc(20% - 22px);
}

.card-item > h4 {
  color: black;
  font-size: 1em;
}

img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: fit-content;
  height: fit-content;
}

.button-left {
  left: 0%;
}

.button-right {
  right: 100%;
}

/* relevant styles */
.img__wrap {
  position: relative;
  height: 200px;
  width: 257px;
  margin-left: auto;
  margin-right: auto;
  overflow: hidden;
}

.img__description {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(29, 106, 154, 0.72);
  color: #fff;
  visibility: hidden;
  height: auto;
  opacity: 0;
  text-align: center;

  /* transition effect. not necessary */
  transition: opacity .2s, visibility .2s;
  margin-top: 0;
  margin-bottom: 0;
}

.img__wrap:hover .img__description {
  visibility: visible;
  opacity: 1;
}
