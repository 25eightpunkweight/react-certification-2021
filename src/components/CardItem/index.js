export { default } from './CardItem.component';