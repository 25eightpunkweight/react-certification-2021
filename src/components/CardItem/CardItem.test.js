import React from 'react'
import {render, cleanup} from '@testing-library/react';
import CardItem from './index.js';
import mockData from './test_data/mock.json';

afterEach(cleanup)

describe ( "render CardItem Component", () => {
   it('should take a snapshot', () => {
      const { asFragment } = render(CardItem(mockData.items[0]));

      expect(asFragment(CardItem(mockData.items[0]))).toMatchSnapshot()
   });

   it ('should show description on hover', () => {
      
   });

});