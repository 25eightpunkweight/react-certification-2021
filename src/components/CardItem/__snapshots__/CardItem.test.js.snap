// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CardItem Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-item"
  >
    <div
      class="card-title"
    >
      <h4>
        Wizeline
      </h4>
    </div>
    <div
      class="img__wrap"
    >
      <img
        class="center"
        src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
      />
      <p
        class="img__description"
      >
        Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
